data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/codec/NewlineDelimitedConfig.java

```diff
@@ -6,12 +6,15 @@
 package com.amazon.dataprepper.plugins.source.codec;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
+
+import java.util.Objects;
 
 /**
  * Configuration class for the newline delimited codec.
  */
 public class NewlineDelimitedConfig {
-    private final int skipLines = 0;
+    private int skipLines = 0;
```

Comment:
I removed the `final` keyword here because I wasn't able to specify > 0 in a YAML file and have the lines skipped properly. I think the unit tests only pass for different skipLines because they mock `getSkipLines`, not because they actually set the skipLines variable (which can't be changed from `0` because of `final`). Removing `final` caused expected behavior. I'm happy to move this issue somewhere else, but I also think it's a minor change and within the scope of the `newline` codec enhancements of this review.

Reply:
Nice catch